packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -0,0 +1,583 @@
+   */
+  rowHeight: PropTypes.string,
+
+  /**
+   * Number indicating the index of the column that represents row header. Index is 0 based and cannot exceed one less than the number of columns in the grid.
+   */
+  rowHeaderIndex: validateRowHeaderIndex,
+
+  /**
+   * Function that is called when a resizable column is resized. Parameters:
+   * @param {string} columnId columnId
+   * @param {string} requestedWidth requestedWidth
+   */
+  onColumnResize: PropTypes.func,
+
+  /**
+   * Callback function that is called when a selectable cell is selected. Parameters:
+   * @param {string} rowId rowId
+   * @param {string} columnId columnId
+   */
+  onCellSelect: PropTypes.func,
+
+  /**
+   * Callback function that is called when a selectable column is selected. Parameters:
+   *  @param {string} columnId columnId
+   */
+  onColumnSelect: PropTypes.func,
+
+  /**
+   * Callback function that is called when all selected cells need to be unselected. Parameters: none.
+   */
+  onClearSelection: PropTypes.func,
+
+  /**
+   * Callback function that is called when a range selection occurs. Parameters:
+   * @param {number} rowIndex rowIndex
+   * @param {number} columnIndex columnIndex
+   * @param {number} direction direction
+   */
+  onRangeSelection: PropTypes.func,
+
+  /**
+   * Boolean indicating whether or not the DataGrid should allow entire rows to be selectable. An additional column will be
+   * rendered to allow for row selection to occur.
+   */
+  hasSelectableRows: PropTypes.bool,
+};
+
+const defaultProps = {
+  rowHeaderIndex: 0,
+  defaultColumnWidth: 200,
+  columnHeaderHeight: '2.5rem',
+  rowHeight: '2.5rem',
+  pinnedColumns: [],
+  overflowColumns: [],
+};
+
+function DataGrid(props) {
+  const {
+    id,
+    ariaLabelledBy,
+    ariaLabel,
+    rows,
+    pinnedColumns,
+    overflowColumns,
+    onColumnResize,
+    defaultColumnWidth,
+    columnHeaderHeight,
+    rowHeight,
+    onColumnSelect,
+    onCellSelect,
+    onClearSelection,
+    onRangeSelection,
+    hasSelectableRows,
+    rowHeaderIndex,
+  } = props;
+
+  if (pinnedColumns.length === 0) {
+    // eslint-disable-next-line no-console
+    console.warn(ERRORS.PINNED_COLUMNS_UNDEFINED);
+  }
+
+  // Default column size constraints
+  const defaultColumnMinimumWidth = 60;
+  const defaultColumnMaximumWidth = 300;
+
+  const [pinnedColumnOffsets, setPinnedColumnOffsets] = useState([0]);
+
+  // Initialize column width properties
+  const initializeColumn = (column) => {
+    const newColumn = { ...column };
+    newColumn.width = column.width || defaultColumnWidth;
+    newColumn.minimumWidth = column.minimumWidth || defaultColumnMinimumWidth;
+    newColumn.maximumWidth = column.maximumWidth || defaultColumnMaximumWidth;
+
+    return newColumn;
+  };
+
+  const displayedColumns = (hasSelectableRows ? [WorklistDataGridUtils.ROW_SELECTION_COLUMN] : []).concat(pinnedColumns).concat(overflowColumns);
```

Comment:
Are there any performance differences in spread vs concat? I think the readability issue would have been fixed if the concats were split off into separate lines.

Reply:
@sdadn Concat is actually more performant. We could even newline this (in which case, we should think about a 
 prettier/eslint config because we currently don't seem to implement maxline limits). I suggested it this way because chaining many concat calls together can make us lose track of what's being tacked onto the array.